src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -922,6 +922,91 @@ CHIP_ERROR DeriveGroupOperationalCredentials(const ByteSpan & epoch_key, const B
     return CHIP_NO_ERROR;
 }
 
+/**
+ * @brief Generate a VendorFabricBindingMessage as used by the Fabric Table Vendor ID Verification Procedure.
+ *
+ * @param[in] fabricBindingVersion - Version of binding payload to generate. outputSpan size requirements are based on this.
+ * @param[in] rootPublicKey - Root public key for the fabric in question
+ * @param[in] fabricId - Fabric ID for the fabric in question
+ * @param[in] vendorId - Vendor ID for the fabric in question
+ * @param[inout] outputSpan - Span that will receive the binding message. Must be large enough for the
+ *                            payload (otherwise CHIP_ERROR_BUFFER_TOO_SMALL) and will be resized to fit.
+ * @return CHIP_NO_ERROR on success, otherwise another CHIP_ERROR value representative of the failure.
+ */
+CHIP_ERROR GenerateVendorFabricBindingMessage(uint8_t fabricBindingVersion, const P256PublicKey & rootPublicKey, FabricId fabricId,
+                                              uint16_t vendorId, MutableByteSpan & outputSpan)
+{
+    // Only V1 supported yet.
+    VerifyOrReturnError(fabricBindingVersion == kFabricBindingVersionV1, CHIP_ERROR_INVALID_ARGUMENT);
+
+    VerifyOrReturnError(outputSpan.size() >= kVendorFabricBindingMessageV1MaxSize, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    Encoding::BigEndian::BufferWriter writer(outputSpan.data(), outputSpan.size());
+
+    // vendor_fabric_binding_message := fabric_binding_version (1 byte) || root_public_key || fabric_id || vendor_id
+    writer.Put8(fabricBindingVersion).Put(rootPublicKey.ConstBytes(), rootPublicKey.Length()).Put64(fabricId).Put16(vendorId);
+
+    size_t actuallyWritten = 0;
+    if (!writer.Fit(actuallyWritten))
```

Comment:
maybe this is also readable:

```cpp
VerifyOrReturnError(writer.Fit(actuallyWritten), CHIP_ERROR_BUFFER_TOO_SMALL);
```

Did not add as suggestion since unsure if this is really more readable. Seems more consistent with verify macros above.